nemo_curator/classifiers/content_type.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from dataclasses import dataclass
+from typing import List, Optional
+
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
+from crossfit.backend.torch.hf.model import HFModel
+from transformers import AutoConfig, AutoTokenizer
+
+from nemo_curator.classifiers.base import (
+    DistributedDataClassifier,
+    HFDeberta,
+    _get_suggest_memory_for_classifier,
+    _run_classifier_helper,
+)
+from nemo_curator.datasets import DocumentDataset
+
+CONTENT_TYPE_IDENTIFIER = "nvidia/content-type-classifier-deberta"
+
+
+@dataclass
+class ContentTypeModelConfig:
+    model: str = "microsoft/deberta-v3-base"
+    fc_dropout: float = 0.2
+    max_len: int = 1024
+
+
+class ContentTypeModel(HFModel):
+    def __init__(
+        self,
+        config: ContentTypeModelConfig,
+        autocast: bool = False,
+        max_mem_gb: Optional[int] = None,
+    ):
+        self.config = config
+        self.autocast = autocast
+        if max_mem_gb is None:
+            max_mem_gb = _get_suggest_memory_for_classifier()
+
+        super().__init__(self.config.model, max_mem_gb=max_mem_gb)
+
+    def load_model(self, device: str = "cuda"):
+        model = HFDeberta.from_pretrained(CONTENT_TYPE_IDENTIFIER)
+        model.set_autocast(self.autocast)
+        model = model.to(device)
+        return model.eval()
+
+    def load_tokenizer(self):
+        return AutoTokenizer.from_pretrained(CONTENT_TYPE_IDENTIFIER)
+
+    def load_config(self):
+        return AutoConfig.from_pretrained(CONTENT_TYPE_IDENTIFIER)
+
+
+class ContentTypeClassifier(DistributedDataClassifier):
+    """
+    ContentTypeClassifier is a text classification model designed to categorize documents into one of 11 distinct speech types based on their content.
+    It analyzes and understands the nuances of textual information, enabling accurate classification across a diverse range of content types.
+    This class is optimized for running on multi-node, multi-GPU setups to enable fast and efficient inference on large datasets.
```

Comment:
```suggestion
    This classifier is optimized for running on multi-node, multi-GPU setups to enable fast and efficient inference on large datasets.
```

Reply:
We should probably add a HF link of something once this model is released so that folks can look at the model card on HF for more details.

Reply:
Thanks! This classifier has been released at https://huggingface.co/nvidia/content-type-classifier-deberta, added the link.